cli-plugins/plugin/plugin.go

```diff
@@ -67,18 +61,18 @@ func RunPlugin(dockerCli *command.DockerCli, plugin *cobra.Command, meta manager
 }
 
 // Run is the top-level entry point to the CLI plugin framework. It should be called from your plugin's `main()` function.
-func Run(makeCmd func(command.Cli) *cobra.Command, meta manager.Metadata) {
+func Run(ctx context.Context, makeCmd func(command.Cli) *cobra.Command, meta manager.Metadata) {
 	otel.SetErrorHandler(debug.OTELErrorHandler)
 
-	dockerCli, err := command.NewDockerCli()
+	dockerCli, err := command.NewDockerCli(command.WithBaseContext(ctx))
```

Comment:
Do plugins add their own ctx to the dockerCli with the `makeCmd`?

Reply:
compose seems to use whatever cobra's context is (https://github.com/docker/compose/blob/85567ae09246ea79191e35f060d230aa5b7c0e5b/cmd/compose/compose.go#L384)

buildx uses this func defined in buildkit though, but watch out since it already handles signals (https://github.com/moby/buildkit/blob/master/util/appcontext/appcontext.go)

Reply:
I see, yeah this will be breaking for plugins. I'll see what I can do to work around that.  

>but watch out since it already handles signals

that's fine, this PR doesn't do any signal handling here, but this is a good point and will work around this